packages/docusaurus-plugin-content-docs/src/cli.ts

```diff
@@ -20,6 +20,67 @@ import {
 import {loadSidebars} from './sidebars';
 import {DEFAULT_PLUGIN_ID} from '@docusaurus/core/lib/constants';
 
+function createVersionedSidebarFile({
+  siteDir,
+  pluginId,
+  sidebarPath,
+  version,
+}: {
+  siteDir: string;
+  pluginId: string;
+  sidebarPath: string | false | undefined;
+  version: string;
+}) {
+  // Load current sidebar and create a new versioned sidebars file (if needed).
+  const loadedSidebars = loadSidebars(sidebarPath);
+
+  // Do not create a useless versioned sidebars file if sidebars file is empty or sidebars are disabled/false)
+  const shouldCreateVersionedSidebarFile =
+    Object.keys(loadedSidebars).length > 0;
```

Comment:
this is the important change: should we create a file in versioned-sidebars folder if the user has disabled the sidebars or is using an empty sidebars.json? 

It used to not be created in some cases so for retrocompatibility I think it make sense to keep not creating it if it would lead to an empty file